firmware/application/ui_navigation.cpp

```diff
@@ -555,6 +560,76 @@ void SystemStatusView::rtc_battery_workaround() {
     }
 }
 
+void SystemStatusView::new_sdcard_structure_checker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path resources_dir = u"SYS";
+    std::vector<std::filesystem::path> directories = scan_root_directories(root_dir);
+
+    auto scan_result = std::find(directories.begin(), directories.end(), resources_dir);
+
+    auto sd_status = sd_card::status();
+
+    if (
+        nav_.is_valid() &&
+        (scan_result == directories.end()) &&
+        (sd_status == sd_card::Status::Mounted)) {
+        nav_.display_modal(
+            "Warning",
+            "        WARNING!!!\n"
+            "SD Card content changed\n"
+            "in this update,\n"
+            "Press YES to auto move.\n"
+            "After finished,\n"
+            "you should move your own\n"
+            "files back into\n"
+            "original folders,\n"
+            "which are now only for users.\n"
+            "You can also choose NO,\n"
+            "and then manually merge\n"
+            "following the Wiki.\n"
+            "Details:\n"
+            "Mayhem wiki - Updating",
+            COMPACTYESNO,
+            [this](bool choice) {
+                if (choice) {
+                    new_sdcard_structure_worker();
+                }
+            });
+    }
+}
+
+void SystemStatusView::new_sdcard_structure_worker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path system_dir = u"SYS";
+
+    std::vector<std::filesystem::path> ignore_dirs = {u"FIRMWARE",
```

Comment:
I had commented earlier that I believe this should be a whitelist (directories to move, versus list of directories not to move).  Otherwise you could move users' personal folders into SYS that don't belong there.  This is especially the case where the user might be using the same SD card for other purposes.  Like I may have used this same SD card in my camera and have photos on there; I don't really want them auto-moved to the SYS folder and have to move them back manually.

Reply:
Would you mind test the current logic for me, to make sure it’s correct? Then I can do whitelist, if you want it appear in this PR.